include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -887,7 +888,41 @@ namespace CUDAWrappers
 
     if (begin != end)
       {
-        if (additional_data.use_coloring)
+        if(additional_data.use_ghost_coloring)
+          {
+            graph.resize(3, std::vector<CellFilter>());
+            
+            std::vector<bool> ghost_vertices(dof_handler.get_triangulation ().n_vertices (), false);
+            
+            for(const auto cell : dof_handler.get_triangulation ().active_cell_iterators ())
+              if(cell->is_ghost())
+                for(unsigned int i = 0; i < GeometryInfo<dim>::vertices_per_cell; i++)
+                  ghost_vertices[cell->vertex_index (i)] = true;
+              
+            std::vector<dealii::FilteredIterator<dealii::TriaActiveIterator<dealii::DoFCellAccessor<dealii::DoFHandler<dim, dim>, false>>>> inner_cells;
+              
+            for (auto cell = begin; cell != end; ++cell)
+            {
+              bool flag = false;
+                
+              for(unsigned int i = 0; i < GeometryInfo<dim>::vertices_per_cell; i++)
+                if(cell->vertex_index (i))
```

Comment:
This is wrong ;) I have to replace it with `if(ghost_vertices[cell->vertex_index (i)])`!

Reply:
Fixing this bug break step-64 :(